zebra-utils/src/bin/zebra-checkpoints/main.rs

```diff
@@ -58,87 +125,111 @@ fn cmd_output(cmd: &mut std::process::Command) -> Result<String> {
+    // zcash-cli returns raw strings without JSON type info.
+    // As a workaround, assume that invalid responses are strings.
+    let response: Value = serde_json::from_str(&response)
+        .unwrap_or_else(|_error| Value::String(response.trim().to_string()));
+
+    Ok(response)
 }
 
 /// Process entry point for `zebra-checkpoints`
+#[tokio::main]
 #[allow(clippy::print_stdout)]
-fn main() -> Result<()> {
+async fn main() -> Result<()> {
     // initialise
     init_tracing();
     color_eyre::install()?;
 
-    // get the current block count
-    let mut cmd = passthrough_cmd();
-    cmd.arg("getblockchaininfo");
+    let args = args::Args::from_args();
 
-    let output = cmd_output(&mut cmd)?;
-    let get_block_chain_info: Value = serde_json::from_str(&output)?;
+    // get the current block count
+    let get_block_chain_info = rpc_output(&args, "getblockchaininfo", None)
+        .await
+        .with_suggestion(|| {
+            "Is the RPC server address and port correct? Is authentication configured correctly?"
+        })?;
 
     // calculate the maximum height
-    let height_limit = block::Height(get_block_chain_info["blocks"].as_u64().unwrap() as u32);
+    let height_limit = get_block_chain_info["blocks"]
+        .try_into_height()
+        .expect("height: unexpected invalid value, missing field, or field type");
 
-    assert!(height_limit <= block::Height::MAX);
     // Checkpoints must be on the main chain, so we skip blocks that are within the
     // Zcash reorg limit.
     let height_limit = height_limit
-        .0
-        .checked_sub(MIN_TRANSPARENT_COINBASE_MATURITY)
-        .map(block::Height)
-        .expect("zcashd has some mature blocks: wait for zcashd to sync more blocks");
-
-    let starting_height = args::Args::from_args().last_checkpoint.map(block::Height);
-    if starting_height.is_some() {
-        // Since we're about to add 1, height needs to be strictly less than the maximum
-        assert!(starting_height.unwrap() < block::Height::MAX);
-    }
+        - HeightDiff::try_from(MIN_TRANSPARENT_COINBASE_MATURITY).expect("constant fits in i32");
+    let height_limit =
+        height_limit.expect("node has some mature blocks: wait for it to sync more blocks");
+
     // Start at the next block after the last checkpoint.
     // If there is no last checkpoint, start at genesis (height 0).
-    let starting_height = starting_height.map_or(0, |block::Height(h)| h + 1);
+    let starting_height = if let Some(last_checkpoint) = args.last_checkpoint {
+        (last_checkpoint + 1)
+            .expect("invalid last checkpoint height, must be less than the max height")
+    } else {
+        Height::MIN
+    };
 
     assert!(
-        starting_height < height_limit.0,
-        "No mature blocks after the last checkpoint: wait for zcashd to sync more blocks"
+        starting_height < height_limit,
+        "No mature blocks after the last checkpoint: wait for node to sync more blocks"
```

Comment:
###### Optional

What about something like

```suggestion
        "The height of the last checkpoint must be less than the height of the last mature block."
```